src/SelfAssessment/AssessmentQuestion.js

```diff
@@ -63,28 +63,29 @@ export const AssessmentQuestion = ({ onNext, onChange, option, question }) => {
 
   const options =
     displayAsOption &&
-    option.values.map((option, index) => (
-      <AssessmentOption
-        answer={selectedValues.find((v) => v.index === index)}
-        index={index}
-        key={option.value}
-        onSelect={(value) => onSelectHandler(value, index)}
-        option={option}
-        isSelected={selectedValues.some((v) => v.index === index)}
-        optionType={question.screen_type}
-      />
-    ))
+    option.values.reduce((result, option, index) => {
```

Comment:
How do you feel about extracting this bit into a method, perhaps in this same file but outside the view logic? Reading it inside this context feels a bit more complex than what it is IMO.